allennlp/interpret/attackers/input_reduction.py

```diff
@@ -0,0 +1,144 @@
+from typing import List
+import numpy as np
+import torch
+from allennlp.interpret.attackers import Attacker
+from allennlp.common.util import JsonDict, sanitize
+from allennlp.data import Instance
+
+@Attacker.register('input-reduction')
+class InputReduction(Attacker):
+    """
+    Runs the input reduction method from `Pathologies of Neural Models Make Interpretations
+    Difficult` https://arxiv.org/abs/1804.07781, which removes as many words as possible
+    from the input _without_ changing the model's prediction.
```

Comment:
Sphinx is not going to be happy with this.  It's RST, not markdown.  Surround text in single backticks for emphasis.